extensions/sql-database-projects/src/models/options/deployOptionsModel.ts

```diff
@@ -0,0 +1,257 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as mssql from 'mssql';
+
+export class DeployOptionsModel {
+	public deploymentOptions: mssql.DeploymentOptions;
+
+	public optionsLookup: Record<string, boolean> = {};
+	public optionsMapTable: Record<string, mssql.DacDeployOptionPropertyBoolean> = {};
```

Comment:
was it not possible for STS to return a map with all these mappings? Schema compare has a similar class that does this, so it would be nice if this logic isn't in two extensions

Reply:
I was able to move the optionsMapTable to STS(table with key:displayoptions and value:value/description). everything works good except updating the Map table with new checkbox value. working on finding a way to fix the error "The left-hand side of an assignment expression may not be an optional property typescript" - I'm facing an issue where mapTableValue?.booleanValue = true, it is only allowing to compare them '==' or '===' not setting it. Will update these unresolved changes once i get it! Thanks

Reply:
As discussed I'm creating new PR for MapTable returning from STS, as this change also involves changing the code on SchemaCompare extension, Thanks :)